pkgs/by-name/ke/keyboard-layout-editor/package.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  gradle_8,
+  makeBinaryWrapper,
+  openjdk,
+  wrapGAppsHook3,
+  xorg,
+}:
+
+let
+  jdk = openjdk.override {
+    enableJavaFX = true;
+  };
+
+in
+stdenv.mkDerivation rec {
+  pname = "keyboard-layout-editor";
+  version = "0-unstable-2019-05-14";
+
+  src = fetchFromGitHub {
+    owner = "vgresak";
+    repo = "keyboard-layout-editor";
+    rev = "308c62ac4bb9ad25429f4d09c0aaa9f72d4194c9";
+    hash = "sha256-ieXHgXAdC1xVJWXwpf9sQ9FaSLeaJfd+Rr7uD8blwHE=";
+  };
+
+  patches = [ ./fix-build-gradle.patch ];
+
+  nativeBuildInputs = [
+    gradle_8
+    makeBinaryWrapper
+    wrapGAppsHook3
+  ];
+
+  mitmCache = gradle_8.fetchDeps {
+    inherit pname;
+    data = ./deps.json;
+  };
+
+  __darwinAllowLocalNetworking = true;
+
+  gradleBuildTask = "fatJar";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/java
+    cp build/libs/keyboard-layout-editor-all-1.0-SNAPSHOT.jar $out/share/java/keyboard-layout-editor.jar
+    mkdir -p $out/bin
+    makeWrapper ${lib.getExe jdk} $out/bin/keyboard-layout-editor \
+      --prefix PATH : ${lib.makeBinPath [ xorg.xkbcomp ]} \
+      --add-flags "--add-opens=javafx.graphics/com.sun.prism=ALL-UNNAMED" \
+      --add-flags "--add-opens=javafx.graphics/com.sun.javafx.font=ALL-UNNAMED" \
+      --add-flags "--add-opens=javafx.graphics/com.sun.javafx.tk=ALL-UNNAMED" \
+      --add-flags "-cp $out/share/java/keyboard-layout-editor.jar cz.gresak.keyboardeditor.Main" \
+      ''${gappsWrapperArgs[@]}
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Keyboard layout editor for XKB";
+    homepage = "https://github.com/vgresak/keyboard-layout-editor";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ fliegendewurst ];
+    mainProgram = "keyboard-layout-editor";
+    platforms = jdk.meta.platforms;
```

Comment:
Looks broken on aarch64-linux
```
error: builder for '/nix/store/plk833gv2sv7ysyyla450jrmgfqd0ary-keyboard-layout-editor-0-unstable-2019-05-14.drv' failed with exit code 1;
       last 25 log lines:
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-fxml/17/javafx-fxml-17-linux-aarch64.jar
       >    > Could not find javafx-controls-17-linux-aarch64.jar (org.openjfx:javafx-controls:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-controls/17/javafx-controls-17-linux-aarch64.jar
       >    > Could not find javafx-graphics-17-linux-aarch64.jar (org.openjfx:javafx-graphics:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-graphics/17/javafx-graphics-17-linux-aarch64.jar
       >    > Could not find javafx-base-17-linux-aarch64.jar (org.openjfx:javafx-base:17).
       >      Searched in the following locations:
       >          https://repo.maven.apache.org/maven2/org/openjfx/javafx-base/17/javafx-base-17-linux-aarch64.jar
       >
       > * Try:
       > > Run with --stacktrace option to get the stack trace.
       > > Run with --info or --debug option to get more log output.
       > > Run with --scan to get full insights.
       > > Get more help at https://help.gradle.org.
       >
       > Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
       >
       > You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
       >
       > For more on this, please refer to https://docs.gradle.org/8.10.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
       >
       > BUILD FAILED in 10s
       For full logs, run 'nix log /nix/store/plk833gv2sv7ysyyla450jrmgfqd0ary-keyboard-layout-editor-0-unstable-2019-05-14.drv'.
```

Reply:
Yeah, I asked about this on the JVM Matrix. The problem is that the resolved dependencies depend on the platform. So I'd have to combine all the deps.json. Another suggestion was to somehow use nixpkgs' openjfx package.

Reply:
Adding `broken = stdenv.hostPlatform.isAarch64;` might be good enough...